.gitlab-ci.yml

```diff
@@ -41,6 +41,31 @@ build:docker:
     - when: never
   script: ["true"]
 
+build:archive:
+  rules:
+    - when: manual
+      allow_failure: true
+  image: "golang:${GO_VERSION}"
+  stage: build
+  script:
+    - export VERSION=$(git describe --tags --dirty)
```

Comment:
CI can sometimes reuse the workspace from another job and it could have a shallow clone with no tags. To be sure this does not hit us later add:
```
  variables:
    GIT_STRATEGY: clone # clone entire repo instead of reusing workspace
    GIT_DEPTH: 0 # avoid shallow clone, this test requires full git history
```